pandas/indexes/base.py

```diff
@@ -1967,8 +1992,33 @@ def symmetric_difference(self, other, result_name=None):
         if result_name is None:
             result_name = result_name_update
 
-        the_diff = sorted(set((self.difference(other)).
-                              union(other.difference(self))))
+        this = self._get_unique_index()
+
+        # Dropping NaN's is a  hack for NaN's backward compatibility
+        # (see the comment in `Index.difference`).
+        dropna = not isinstance(self, ABCMultiIndex) and \
```

Comment:
use same idiom as above. you *may* want to factor out some of this to a shared (private) method for use in difference as well.